components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/ValidateUsernameApiServiceImpl.java

```diff
@@ -43,80 +48,185 @@
 
     private static final Log LOG = LogFactory.getLog(ResendCodeApiServiceImpl.class);
     private static final String SKIP_SIGN_UP_ENABLE_CHECK_KEY = "skipSignUpEnableCheck";
+    private static final String USERNAME_JAVA_REG_EX_VIOLATION_ERROR_MSG = "UsernameJavaRegExViolationErrorMsg";
+    private static final String DISABLE_USER_NAME_VALIDITY_API_ERROR_RESPONSES =
+            "DisableUserNameValidityApiErrorResponses";
 
     @Override
     public Response validateUsernamePost(UsernameValidationRequestDTO user) {
 
+        // Checking the property DisableUserNameValidityApiErrorResponses in identity.xml to decide whether to use the
+        // newly improved method or the old one.
+
+        if (isUserNameValidityApiErrorResponsesDisabled()) {
```

Comment:
We don't need to check the property in the public fix. We only checked that in the other fix to maintain backward compatibility. Since we're correcting an erroneous behavior, we can skip this check.